pkg/kev/services.go

```diff
@@ -135,24 +135,25 @@ func (sc ServiceConfig) validate() error {
 	}
 
 	if !result.Valid() {
-		// Prioritise clear error messages.
+		// If we don't find anything useful just go with whatever is available.
+		err := errors.New(result.Errors()[0].Description())
+
 		for _, e := range result.Errors() {
+			// Prioritise clear error messages.
 			if e.Type() == "required" {
 				return errors.New(e.Description())
 			}
-		}
 
-		for _, e := range result.Errors() {
 			// These errors are very cryptic and hurt usability.
 			if e.Type() == "number_one_of" || e.Type() == "number_any_of" || e.Type() == "number_all_of" {
 				continue
 			}
 
-			return errors.New(e.Description())
+			// this is a valid error if there isn't anything better
+			err = errors.New(e.Description())
```

Comment:
Don't you need to return here? The above will return the last error message - was it an intention?

Reply:
this is what makes this thing weird, if you get to this line you still have no guarantee the one of the next iterations isn't a "required" error. so we want to make sure we keep one "good enough" error around in case there ins't a required. Does that make sense?

Reply:
Feels there should be a better way to address the above.

- Perhaps first finding an error with `e.Type() == "required"` ... maybe using `go-funk` -> `funk.Find(...)
- And then finding any error that is NOT `e.Type() == "number_one_of" || e.Type() == "number_any_of" || e.Type() == "number_all_of"` and returning.

See https://github.com/thoas/go-funk

Reply:
Find will do the exact same thing my original code did which is basically iterate and check if there is a match to the predicate. And then that's exactly what I was doing, not sure there is a point in introducing a library to avoid writing a for-loop?